### ISSUE TITLE
Pydantic Datetime parsing, string cannot be assigned Pylance error

### ISSUE THREAD

**Issue by JonathanNakandala** [microsoft/pyright#3678]
**Describe the bug**

I followed the description of how Pydantic does the parsing to datetime objects automatically:

https://pydantic-docs.helpmanual.io/usage/types/#datetime-types

However there's this linting error over the model inputs.

```
Argument of type "Literal['2022-06-07T16:30:54.000Z']" cannot be assigned to parameter "timestamp" of type "datetime" in function "__init__"
  "Literal['2022-06-07T16:30:54.000Z']" is incompatible with "datetime"Pylance[reportGeneralTypeIssues](https://github.com/microsoft/pylance-release/blob/main/DIAGNOSTIC_SEVERITY_RULES.md#diagnostic-severity-rules)
```

**To Reproduce**
Paste the following code in to a python environment with pydantic installed
```
from datetime import datetime
from pydantic import BaseModel

class MyData(BaseModel):
    timestamp: datetime

data = MyData(timestamp="2022-06-07T16:30:54.000Z")
print(data.dict())

```

The output is:

```python
{'timestamp': datetime.datetime(2022, 6, 7, 16, 30, 54, tzinfo=datetime.timezone.utc)}
```

**Expected behavior**

Perhaps I'm missing something to make Pylance happy?

Or is this more of an esoteric implementation by Pydantic that isn't really compliant to Python?

**Screenshots or Code**
Python Version: 3.10.4
Pydantic Version 1.9.1

**VS Code extension or command-line**
VS Code Version: 1.69.0 
Pylance Version: v2022.7.20

**Comment by erictraut** [comment]
Pyright's behavior is correct here. You are attempting to assign a `str` value to an argument that has a type `datetime`. If pydantic is performing an under-the-cover conversion from `str` to `datetime`, pyright would have no way of knowing that. 

To make this work with pyright, you'd need to call the appropriate conversion routine and convert the `str` to `datetime` yourself.

**Comment by drorata** [comment]
While I totally understand the point made by @erictraut, I still find this a counter-productive situation. On one hand, `pydantic` is extremely popular and useful. A common pattern with `pydantic` is the one described in the original post. Yes, `pydantic` does do some conversions under the hood and they are very much desired. On the other hand, `pylance` is also very popular and helpful. 

Having said that, I don't think it is reasonable to implement the casts from `str` to `datetime` outside of `pydantic`; it undermines a lot of the usefulness of the latter. It is also not productive to add `# type: ignore` wherever an instance is instantiated with strings.

I wonder whether there are any updates on this topic.
